stata_kernel/code_manager.py

```diff
@@ -8,18 +8,43 @@ class CodeManager(object):
     """Class to deal with text before sending to Stata
     """
 
-    def __init__(self, code, semicolon_delimit=False):
+    def __init__(self, code, semicolon_delimit=False, mata_mode=False):
+
+        # mata goes first since it obeys #delimit!
         if semicolon_delimit:
             code = '#delimit ;\n' + code
+            if mata_mode:
+                code = 'mata;\n' + code
+        elif mata_mode:
+            code = 'mata\n' + code
+
+        # print('debugz0', code)
         self.input = code
         self.tokens = self.tokenize_code(code)
+        # print('debugz1', self.tokens)
         self.tokens_nocomments = self.remove_comments()
+        # print('debugz2', self.tokens_nocomments)
```

Comment:
The init to code_manager may be changed a bit. See 
https://github.com/kylebarron/stata_kernel/tree/dev-two-lexers

Reply:
Should I merge dev-two-lexers and continue from there?

Reply:
I'd say so. I had to refactor a bit due to splitting into two lexers.

Reply:
Sometimes I feel like git is magic. Anyway, I merged in the dev-two-lexers branch. Same problems as before, tho. Hopefully with the new paradigm they will be easier to solve.

Reply:
Which same problems?

Reply:
Mainly lack of `#delimit ;` support in mata and `if` trips it up (because in mata, if by itself does not finish execution. It wants an "else" statement).

Reply:
Sounds like we need a Mata lexer as well...

Reply:
No, no. I think I can do it with the right nesting scheme.

Reply:
More generally, there are actually several instances where a line can be sent to mata, and the reply from stata can just be a line continuation prompt...

Reply:
That sounds tough. Any way to force a dot prompt?